bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,408 @@
+                    # subtract one, since one reaction was done by the bot
+
+                logger.debug(f"{msg.author} got a score of {score}")
+                data = json.loads(await self.messages.get(message_id))
+                data["score"] = score
+                await self.messages.set(message_id, json.dumps(data))
+
+            # Sort the winner messages
+            winner_messages = sorted(
+                [(msg_id, json.loads(usr_data)) for msg_id, usr_data in await self.messages.items()],
+                key=lambda x: x[1]["score"],
+                reverse=True,
+            )
+
+            winners = []
+            for i, winner in enumerate(winner_messages):
+                winners.append(winner)
+                if len(winner_messages) > i + 1:
+                    if winner_messages[i + 1][1]["score"] != winner[1]["score"]:
+                        break
+                elif len(winner_messages) == (i + 1) + 1:  # The next element is the last element
+                    if winner_messages[i + 1][1]["score"] != winner[1]["score"]:
+                        break
+
+            # one iteration is complete
+            await channel.send("Today's Spook Name Rate Game ends now, and the winner(s) is(are)...")
+
+            async with channel.typing():
+                await asyncio.sleep(1)  # give the drum roll feel
+
+                if not winners:  # There are no winners (no participants)
+                    await channel.send("Hmm... Looks like no one participated! :cry:")
+                    return
+
+                if len(winners) > 1:
+                    await channel.send(" and ".join(PING.format(id=win[1]["author"]) for win in winners) + " !")
+                    score = winners[0][1]["score"]
+                    # display winners, their names and scores
+                    await channel.send(f"Congratulations to all! You have a score of {score}!")
+                    names = ", ".join([f'{win[1]["name"]} ({PING.format(id=win[1]["author"])})' for win in winners])
+                    await channel.send(f"Your names were: {names}!")
+
+                else:  # there is only one winner
+                    # There cant be 0 winners since we are checking that above
+
+                    winner = winners[0][1]
+                    await channel.send(f"{PING.format(id=winner['author'])}!")
+                    await channel.send(
+                        f"Congratulations {PING.format(id=winner['author'])}! You have a score of "
+                        f"{winner['score']}!"
+                    )
+                    await channel.send(f"Your name was: **{winner['name']}**.")  # display winner, name and score
+
+                # Send random party emojis
+                party = [random.choice([":partying_face:", ":tada:"]) for _ in range(random.randint(1, 10))]
+                await channel.send(" ".join(party))
```

Comment:
```suggestion
                party = (random.choice([":partying_face:", ":tada:"]) for _ in range(random.randint(1, 10)))
                await channel.send(" ".join(party))
```
Don't need the temporary list